src/components/Frame/components/Loading/Loading.tsx

```diff
@@ -1,78 +1,69 @@
-import React from 'react';
+import React, {useEffect, useState} from 'react';
 import debounce from 'lodash/debounce';
+import {useLazyRef} from '../../../../utilities/use-lazy-ref';
 import styles from './Loading.scss';
 
-export interface LoadingProps {}
-
-interface State {
-  progress: number;
-  step: number;
-  animation: number | null;
-}
-
 const INITIAL_STEP = 10;
 const STUCK_THRESHOLD = 99;
 
-export class Loading extends React.Component<LoadingProps, State> {
-  state: State = {
-    progress: 0,
-    step: INITIAL_STEP,
-    animation: null,
-  };
-
-  private ariaValuenow = debounce(() => {
-    const {progress} = this.state;
-    return Math.floor(progress / 10) * 10;
-  }, 15);
+export interface LoadingProps {}
 
-  componentDidMount() {
-    this.increment();
-  }
+export function Loading(_props: LoadingProps) {
+  const [progress, setProgress] = useState(0);
 
-  componentWillUnmount() {
-    const {animation} = this.state;
+  useEffect(() => {
+    let animation: number | undefined;
+    let step = INITIAL_STEP;
+    let currentProgress = 0;
+    let previusProgress = 0;
 
-    if (animation != null) {
-      cancelAnimationFrame(animation);
-    }
-  }
+    const increment = () => {
+      if (currentProgress >= STUCK_THRESHOLD) {
```

Comment:
You seem to have changed the progress from being an integer between 0 an 100, to a number rounded to 2 decimal places between 0 and 1. I'm ok with that, but make sure that you update the stuck threshold and anything else like how the next step is calculated to account for that.

Reply:
`currentProgress` still go from 0 - 100 with not limited decimal point (e.g. 20, 21.332232332, 22.12321, 23.5565, etc)
`nextProgress` and `previousProgress` goes from 0 - 1 with only two decimal point (e.g. 0.12, 0.13, 0.14, 0.15)
with this approach set state called only when `nextProgress` and `previousProgress` are not equal which save redundant re-rendering. if we set `currentProgress` in state, it'd re-render more often (due its decimals changes with each `requestAnimationFrame` call)